language/changes/7-packages.md

```diff
@@ -0,0 +1,323 @@
+arises in `P`: which "`N`" is meant when `N` is referred to in `P`? The one
+from `P1` or `P2`? To prevent this ambiguity around which package a named
+address is coming from, we enforce that the sets of scopes introduced by all
+dependencies in a package are disjoint, and provide a way to _rename named
+addresses_ when the package that brings them into scope is imported.
+
+Renaming a named address when importing can be done as follows in our `P`,
+`P1`, and `P2` example above:
+
+```
+[package]
+name = "P"
+...
+[dependencies]
+P1 = { local = "some_path_to_P1", addr_subst = { "P1N" = "N" } }
+P2 = { local = "some_path_to_P2"  }
+```
+
+With this renaming `N` refers to the `N` from `P2` and `P1N` will refer to `N`
+coming from `P1`:
+
+```
+module N::A {
+    public fun x(): address { @P1N }
+}
+```
+
+It is important to note that _renaming is not local_: once a named address `N`
+has been renamed to `N'` in a package `P` all packages that import `P` will not
+see `N` but only `N'` unless `N` is reintroduced from outside of `P`. This is
+why rule (2) in the scoping rules at the start of this section specifies a
+"dependency path in the package graph between between `P` and the declaring
+package of `N` with no renaming of `N`."
+
+### Instantiation
+
+Named addresses can be instantiated multiple times across the package graph as
+long as it is always with the same value. It is an error if the same named
+address (regardless of renaming) is instantiated with differing values across
+the package graph.
+
+A Move package can only be compiled if all named addresses resolve to a value.
+This presents issues if the package wishes expose an uninstantiated named
+address. This is what the `[dev-addresses]` section solves. This section can
+set values for named addresses, but cannot introduce any named addresses.
+Additionally, only the `[dev-addresses]` in the root package are included in
+`dev` mode. For example a root package with the following manifest would not compile
+outside of `dev` mode since `NamedAddr` would be uninstantiated:
+
+```
+[package]
+name = "ExamplePkg"
+...
+[addresses]
+NamedAddr = "_"
+
+[dev-addresses]
+NamedAddr = "0xC0FFEE"
+```
+
+## Usage, Artifacts, and Data Structures
+
+The Move package system comes with a command line option as part of the Move
+CLI `move package <package_flags> <command> <command_flags>`. Unless a
+particular path is provided, all package commands will run in the current working
+directory. The full list of commands and flags for the Move Package CLI can be found by
+running `move package --help`.
+
+### Usage
+
+A package can be compiled either through the Move CLI commands, or as a library
+command in Rust with the function `compile_package`. This will create a
+`CompiledPackage` that holds the compiled bytecode along with other compilation
+artifacts (source maps, documentation, ABIs) in memory. This `CompiledPackage`
+is interconvertible with the `OnDiskPackage` -- the latter being the data of
```

Comment:
"interconvertible"? I don't think that's a word.

Reply:
IMO this is the [perfect word](https://www.merriam-webster.com/dictionary/interconversion) for this, i.e., that the two can be converted back-and-forth, but I'll see if I can find a better phrasing for this to leave this word out.

Reply:
OK, fine, I stand corrected. I still think it sounds a bit strange, but whatever.

Reply:
Phrased it differently and removed it, let me know what you think of the updated text here :)